### ISSUE TITLE
dartanalyzer --help is sent to stderr instead of stdout

### ISSUE THREAD

**Issue by pq** [dart-lang/sdk#25133]
_From @alextekartik on November 5, 2015 22:31_

Tested on Ubuntu 15.10 with dart version 1.13.0-dev.7.9
Help output is sent to stderr (was sent to stdout in 1.12.2)

```
$ dartanalyzer --help 2>/dev/null
```

gives nothing. Other tools work fine though...

```
$ dartfmt --help 2>/dev/null
Reformats whitespace in Dart source files.
...
```

Standard seems to be stdout though: http://www.gnu.org/prep/standards/html_node/_002d_002dhelp.html
Not a big deal but I actually have a unit test for wrapper around dart command line tools:
https://github.com/alextekartik/tekartik_io_tools.dart/blob/master/test/dartbin_utils_test.dart#L52-L55

_Copied from original issue: dart-lang/analyzer_cli#76_

**Comment by pq** [comment]
De-scoping.

**Comment by jamesderlin** [comment]
While this isn't critical, I consider this to be bad.  It means that I can't do `dartanalyzer --help | less` to get readable, paginated help output (particular with `--help --verbose`).

I know that I instead can do `dartanalyzer --help 2>&1 | less`, but I shouldn't have to.  Proper command-line programs should not treat `-h` and `--help` as errors.

**Comment by srawlins** [comment]
Somehow this was partly fixed? `dartanalyzer --help |less` pagination works, but `dartanalyzer --help` still prints to stderr.

FWIW, `dart --help` is printed to stderr as well. I can't see this ever being high-priority enough to break existing clients...

**Comment by jamesderlin** [comment]
> Somehow this was partly fixed? `dartanalyzer --help |less` pagination works, but `dartanalyzer --help` still prints to stderr.

`dartanalyzer --help | less` does not work for me, and I don't see how it could work if it's still printing to stderr.

> FWIW, `dart --help` is printed to stderr as well. I can't see this ever being high-priority enough to break existing clients...

1. According to https://github.com/dart-lang/sdk/issues/25133#issue-120806620, we use to print to stdout, so we already broke clients.  This would be fixing a regression.

2. The existing behavior is rather user-hostile.

3. End-users accustomed to doing `dartanalyzer --help 2>&1 | less` wouldn't be broken.

4. Scripts inspecting the `--help` text might be broken, but I don't see much utility in running `--help` from a script, so I'd expect such cases to be rare. (The linked unit test in https://github.com/dart-lang/sdk/issues/25133#issue-120806620 no longer seems to exist.  Not sure if it moved.) IMO any such scripts should expect to be brittle anyway since help text could easily change.

**Comment by srawlins** [comment]
All good points!

**Comment by jamesderlin** [comment]
It looks like this will no longer be a problem with 2.9's unified `dart` tool anyway. =)

**Comment by srawlins** [comment]
Agreed. `dartanalyzer` is being deprecated in favor of `dart analyze`.
